src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs

```diff
@@ -0,0 +1,239 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+using Aspire.Hosting.Dcp.Process;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Globalization;
+using Aspire.Hosting.Properties;
+
+namespace Aspire.Hosting.Dcp;
+
+internal sealed partial class DcpDependencyCheck
+{
+    // Docker goes to into resource saver mode after 5 minutes of not running a container (by default).
+    // While in this mode, the commands we use for the docker runtime checks can take quite some time
+    private const int WaitTimeForDcpInfoCommandInSeconds = 10;
+
+    [GeneratedRegex("[^\\d\\.].*$")]
+    private static partial Regex VersionRegex();
+
+    public static async Task EnsureDcpDependenciesAsync(string? dcpPath, bool hasContainers, string? containerRuntime, CancellationToken cancellationToken)
+    {
+        if (!File.Exists(dcpPath))
+        {
+            throw new FileNotFoundException($"The Aspire application host is not installed at \"{dcpPath}\". The application cannot be run without it.", dcpPath);
+        }
+
+        try
+        {
+            await EnsureDcpVersionAsync(dcpPath, cancellationToken).ConfigureAwait(false);
+            await EnsureDcpContainerRuntimeAsync(dcpPath, hasContainers, containerRuntime, cancellationToken).ConfigureAwait(false);
+        }
+        catch (Exception ex) when (ex is not DistributedApplicationException)
+        {
+            Console.Error.WriteLine(string.Format(
+                CultureInfo.InvariantCulture,
+                Resources.DcpDependencyCheckFailedMessage,
+                ex.ToString()
+            ));
+        }
+    }
+
+    private static async Task EnsureDcpVersionAsync(string dcpPath, CancellationToken cancellationToken)
```

Comment:
Perhaps DCP should be modified to output the version number with the `info` command so we only need to launch the process once?

Reply:
Ha, you beat me to it! ☺️

Reply:
It does and that’s the way I initially implemented this check, but DCP versions before 0.1.51 don’t have the info command, which means we error in a less than helpful way on old workloads. I thought about treating that as version not supported, but it felt a bit messy.

Reply:
I think that we can rebaseline  the version of DCP we work with for each preview :) Given it is in the workload.

Reply:
There are some tweaks related to the new pinned workload sets that should make it a lot harder for the AppModel and DCP to drift versions in the long run. I've switched back to a single call to Info and will just treat errors as an unsupported version.